jme3-core/src/main/resources/Common/ShaderLib/PBR.glsllib

```diff
@@ -35,7 +51,11 @@ vec3 sphericalHarmonics( const in vec3 normal, const vec3 sph[9] ){
 }
 
 
-float PBR_ComputeDirectLight(vec3 normal, vec3 lightDir, vec3 viewDir,
+float PBR_ComputeDirectLight(
```

Comment:
Could you please  refactor  this to a function that accepts a precalculated alpha  eg. `_PBR_ComputeDirectLight(in float alpha,. ..`
and then have two more functions 
- `PBR_ComputeDirectLight` that calls `_PBR_ComputeDirectLight` with `alpha=roughness*roughness` (the current non antialiased bahavior)

and 

- `PBR_ComputeDirectLightAA` that calls `_PBR_ComputeDirectLight` using the antialiasing corrected alpha (new method)

and then call PBR_ComputeDirectLightAA if SPECULAR_AA is defined or PBR_ComputeDirectLight otherwise.

In this way we let the shaders implementing PBR.glsllib deal with their own way to detect if antialiasing is required, instead of forcing them to defined SPECULAR_AA

Reply:
Good idea. I think no need to define two PBR_ComputeDirectLight and PBR_ComputeDirectLightAA, just adjust the roughness parameter to alpha as function parameter, then calculate externally via
#SPECULAR_AA
alpha = Specular_Anti_Aliasing();
#else
alpha = roughness * roughness;
#endif
PBR_ComputeDirectLight(，，alpha，...);
What do you think?

Reply:
> I think the #SPECULAR_AA macro definition belongs to material parameters and should not appear in PBR.glsllib

Indeed, that's what i was proposing, to remove the macro from PBR.glsllib, however the rationale behind keeping the signature of PBR_ComputeDirectLight unchanged and to add another function PBR_ComputeDirectLightAA is to maintain backward compatibility.
PBR_ComputeDirectLightAA doesn't need to check if specular AA is enabled internally, since PBRLighting etc would do the check and call it only if specular AA is enabled.

Reply:
> > I think the #SPECULAR_AA macro definition belongs to material parameters and should not appear in PBR.glsllib
> 
> Indeed, that's what i was proposing, to remove the macro from PBR.glsllib, however the rationale behind keeping the signature of PBR_ComputeDirectLight unchanged and to add another function PBR_ComputeDirectLightAA is to maintain backward compatibility. PBR_ComputeDirectLightAA doesn't need to check if specular AA is enabled internally, since PBRLighting etc would do the check and call it only if specular AA is enabled.

I roughly understand what you said. Considering backwards compatibility, I will adjust the code here.

Reply:
I have adjusted the related code